sdk/bcs/src/index.ts

```diff
@@ -531,14 +561,33 @@ export class BCS {
    *
    * console.assert(toHex(serialized) === '06010203040506');
    *
-   * @param type Name of the type to serialize (must be registered).
+   * @param type Name of the type to serialize (must be registered) or a struct type.
    * @param data Data to serialize.
    * @param size Serialization buffer size. Default 1024 = 1KB.
    * @return A BCS reader instance. Usually you'd want to call `.toBytes()`
    */
-  public ser(type: string, data: any, size: number = 1024): BcsWriter {
-    let { typeName, typeParams } = this.parseTypeName(type);
-    return this.getTypeInterface(typeName).encode(data, size, typeParams);
+  public ser(
+    type: string | StructTypeDefinition,
+    data: any,
+    size: number = 1024
+  ): BcsWriter {
+    if (typeof type === "string") {
+      let { typeName, typeParams } = this.parseTypeName(type);
+      return this.getTypeInterface(typeName).encode(data, size, typeParams);
+    }
+
+    // Quick serialization without registering the type in the main struct.
+    if (typeof type == "object") {
+      let temp = new BCS(this);
+      temp.registerStructType(this.tempKey, type);
```

Comment:
will it cause any problem if people accidentally register using the same `temp_struct`, should we throw an error when people try to do so?